src/main/java/picard/fingerprint/CrosscheckFingerprints.java

```diff
@@ -139,12 +147,29 @@
     private double[][] crosscheckMatrix = null;
     private final List<String> matrixKeys = new ArrayList<>();
 
+
+
     @Override
     protected int doWork() {
         // Check inputs
-        INPUT.forEach(IOUtil::assertFileIsReadable);
+        IOUtil.assertFilesAreReadable(INPUT);
+        IOUtil.assertFilesAreReadable(INPUT2);
+
         IOUtil.assertFileIsReadable(HAPLOTYPE_MAP);
         if (OUTPUT != null) IOUtil.assertFileIsWritable(OUTPUT);
+
+        if (!INPUT2.isEmpty()) {
```

Comment:
Won't this throw a NPE if the optional INPUT2 is not specified?
Use StringUtils.isEmpty